classes/class-wc-connect-functions.php

```diff
@@ -95,5 +95,130 @@ public static function user_can_manage_labels() {
 		}
+
+		/**
+		 * Exports existing tax rates to a CSV and clears the table.
+		 *
+		 * Ported from TaxJar's plugin.
+		 * See: https://github.com/taxjar/taxjar-woocommerce-plugin/blob/42cd4cd0/taxjar-woocommerce.php#L75
+		 *
+		 * @return boolean
+		 */
+		public static function backup_existing_tax_rates() {
+			global $wpdb;
+
+			// Export Tax Rates
+			$rates = $wpdb->get_results(
+				$wpdb->prepare(
+					"SELECT * FROM {$wpdb->prefix}woocommerce_tax_rates
+			ORDER BY tax_rate_order
+			LIMIT %d, %d
+			",
+					0,
+					10000
+				)
+			);
+
+			ob_start();
+			$header =
+				__( 'Country Code', 'woocommerce' ) . ',' .
+				__( 'State Code', 'woocommerce' ) . ',' .
+				__( 'ZIP/Postcode', 'woocommerce' ) . ',' .
+				__( 'City', 'woocommerce' ) . ',' .
+				__( 'Rate %', 'woocommerce' ) . ',' .
+				__( 'Tax Name', 'woocommerce' ) . ',' .
+				__( 'Priority', 'woocommerce' ) . ',' .
+				__( 'Compound', 'woocommerce' ) . ',' .
+				__( 'Shipping', 'woocommerce' ) . ',' .
+				__( 'Tax Class', 'woocommerce' ) . "\n";
+
+			echo $header;
+
+			foreach ( $rates as $rate ) {
+				if ( $rate->tax_rate_country ) {
+					echo esc_attr( $rate->tax_rate_country );
+				} else {
+					echo '*';
+				}
+
+				echo ',';
+
+				if ( $rate->tax_rate_state ) {
+					echo esc_attr( $rate->tax_rate_state );
+				} else {
+					echo '*';
+				}
+
+				echo ',';
+
+				$locations = $wpdb->get_col( $wpdb->prepare( "SELECT location_code FROM {$wpdb->prefix}woocommerce_tax_rate_locations WHERE location_type='postcode' AND tax_rate_id = %d ORDER BY location_code", $rate->tax_rate_id ) );
+
+				if ( $locations ) {
+					echo esc_attr( implode( '; ', $locations ) );
+				} else {
+					echo '*';
+				}
+
+				echo ',';
+
+				$locations = $wpdb->get_col( $wpdb->prepare( "SELECT location_code FROM {$wpdb->prefix}woocommerce_tax_rate_locations WHERE location_type='city' AND tax_rate_id = %d ORDER BY location_code", $rate->tax_rate_id ) );
+				if ( $locations ) {
+					echo esc_attr( implode( '; ', $locations ) );
+				} else {
+					echo '*';
+				}
+
+				echo ',';
+
+				if ( $rate->tax_rate ) {
+					echo esc_attr( $rate->tax_rate );
+				} else {
+					echo '0';
+				}
+
+				echo ',';
+
+				if ( $rate->tax_rate_name ) {
+					echo esc_attr( $rate->tax_rate_name );
+				} else {
+					echo '*';
+				}
+
+				echo ',';
+
+				if ( $rate->tax_rate_priority ) {
+					echo esc_attr( $rate->tax_rate_priority );
+				} else {
+					echo '1';
+				}
+
+				echo ',';
+
+				if ( $rate->tax_rate_compound ) {
+					echo esc_attr( $rate->tax_rate_compound );
+				} else {
+					echo '0';
+				}
+
+				echo ',';
+
+				if ( $rate->tax_rate_shipping ) {
+					echo esc_attr( $rate->tax_rate_shipping );
+				} else {
+					echo '0';
+				}
+
+				echo ',';
+
+				echo "\n";
+			} // End foreach().
+
+			$csv = ob_get_contents();
+			ob_end_clean();
```

Comment:
```suggestion
			$csv = ob_get_clean();
```
Is there any reason to separate `ob_get_contents()` and `ob_end_clean()`? Perhaps we can make it shorter?